Update endpoint-designer.md (#3832)
TykTechnologies/tyk-docs: tyk-docs/content/advanced-configuration/transform-traffic/endpoint-designer.md

To get started, click **Add Endpoint**, this will give you an empty path definition:

In a new path definition, you can set multiple options, and if you do not specify a specific action for that list (from the plugins drop-down), then saving the path will do nothing (and it will vanish).

Your options are:

- **Method**: The method you are targeting, can be any valid HTTP method, simply pick one from the drop-down menu.
- **Relative Path**: The relative path to the target. For example, if your API is listening on an `/api` listen path, and you want to match the `/api/get` URL, in the Endpoint Designer you should match for the `/get` endpoint. It is important to exclude aberrant slashes (`/`) from your path matching, as otherwise the gateway may not match the path correctly. A path can contain wild cards, such as `{id}`, the actual value in the wildcard is not used (it is translated into a regex), however, it is useful to make the path more human-readable when editing.
- **Plugin**: A path can belong to multiple plug-ins, these plug-ins define the behaviour you want to impose on the matched request.

{{< note success >}}
**Note**  

> **Endpoint parsing**

Author: @ilijabojanovic can you please review only this section?

> With the following plugins (Mock Response, {{<fn>}}Blocklist{{</fn>}} and {{<fn>}}Allowlist{{</fn>}}) adding a `$` at the end of your URL. This ensures that regular expression matching avoids endpoints with characters following the specified endpoint.

Reviewer: ```suggestion
The following plugins (Mock Response, {{<fn>}}Blocklist{{</fn>}} and {{<fn>}}Allowlist{{</fn>}}) require a `$` added at the end of your URL. This ensures that regular expression matching is exact, avoiding endpoints with characters following the specified endpoint.
```

<br/>
Here's an example to illustrate its impact: When you define an `/anything` endpoint, it also implies a potential match for `/anything/somepath`. However, employing `/anything$` in your definition specifically prevents the matching of `/somepath`. This distinction is interpreted as follows:

> - In an allowlist setup, only `/anything` is permitted, and `/anything/somepath` remains blocked unless explicitly added.

Reviewer: ```suggestion
- In an Allowlist setup, only `/anything` is permitted, and `/anything/somepath` remains blocked unless explicitly added.
```

}
```

> #### Case Sensitivity of Allowlist

Author: @ilijabojanovic  here

> By default the {{<fn>}}Allowlist{{</fn>}} endpoint plugin is case-sensitive, so for example if `getuser` is allowed, while `getUser` and `GetUser` are not. If you select the **Ignore Case** option in the {{<fn>}}Allowlist{{</fn>}} plugin settings, all three options will be allowed.

Reviewer: ```suggestion
By default the {{<fn>}}Allowlist{{</fn>}} endpoint plugin is case-sensitive, so for example `getuser` is allowed, while `getUser` and `GetUser` are not. If you select the **Ignore Case** option in the {{<fn>}}Allowlist{{</fn>}} plugin settings, all three options will be allowed.
```

{{< note success >}}
**Note**  

> You can also set a global ignore case on the API level or across [the gateway]({{< ref "tyk-oss-gateway/configuration#ignore_endpoint_case" >}}) in `tyk.conf`. These global settings will override this endpoint-level setting. (Added in v2.9.4).

Author: @ilijabojanovic here
